asapdiscovery-data/asapdiscovery/data/schema_v2/ligand.py

```diff
@@ -0,0 +1,223 @@
+from pathlib import Path
+from typing import Any, Dict, Optional, Tuple, Union  # noqa: F401
+from uuid import UUID
+
+from asapdiscovery.data.openeye import (
+    _get_SD_data_to_object,
+    _set_SD_data_repr,
+    get_SD_data,
+    oechem,
+    oemol_to_inchi,
+    oemol_to_inchikey,
+    oemol_to_sdf_string,
+    oemol_to_smiles,
+    print_SD_Data,
+    sdf_string_to_oemol,
+    set_SD_data,
+    smiles_to_oemol,
+)
+from pydantic import UUID4, Field
+
+from .experimental import ExperimentalCompoundData
+from .schema_base import (
+    DataModelAbstractBase,
+    DataStorageType,
+    read_file_directly,
+    write_file_directly,
+)
+
+
+class InvalidLigandError(ValueError):
+    ...
+
+
+# Ligand Schema
+
+
+class LigandIdentifiers(DataModelAbstractBase):
+    """
+    Identifiers for a Ligand
+    """
+
+    moonshot_compound_id: Optional[str] = Field(
+        None, description="Moonshot compound ID"
+    )
+    manifold_api_id: Optional[UUID] = Field(
+        None, description="Unique ID from Postera Manifold API"
+    )
+    manifold_vc_id: Optional[str] = Field(
+        None, description="Unique VC ID (virtual compound ID) from Postera Manifold"
+    )
+    compchem_id: Optional[UUID4] = Field(
+        None, description="Unique ID for P5 compchem reference, unused for now"
+    )
+
+    def to_SD_tags(self) -> dict[str, str]:
+        """
+        Convert to a dictionary of SD tags
+        """
+        data = self.dict()
+        return {str(k): str(v) for k, v in data.items() if v is not None}
+
+
+class Ligand(DataModelAbstractBase):
```

Comment:
Would it be useful to have some creation provenance here as well? We could store data depending on the class method used to make the molecule for example the function name and the input data?  Something like:
```python
# from smiles
provenance = Provenance(method="from_smiles", input="CCC", version="asap.__version__", date=datetime)
# from sdf
provenance = Provenance(method="from_sdf", input="path/to/file", version="asap.__version__", date=datetime)

```

Reply:
I am going to push provenance to another PR, issue #343